### ISSUE TITLE
Code generation bug for triusolve function calls

### ISSUE THREAD

**Issue by andreasbell** [casadi/casadi#3745]
There seems to be a bug in the code generation for some function calls with triusolve, where the wrong pointer is used to pass in data and store the results of the function call. 

Here is a minimum working example where evaluating the function in the casadi "virual machin", and the compiled generated code give different results.
```python
A = ca.MX.sym("A", 1, 1)
b = ca.MX.sym("b", 1, 2)

x = ca.solve(A, b)

fun = ca.Function("fun", [A, b], [x])

fun.generate()

# Compiling code to shared library and importing it ...
fun_c = ca.external(fun.name(), "fun.dll")

fun(10, [1, 2]) # -> DM([[0.1, 0.2]])
fun_c(10, [1, 2]) # -> DM([[1, 2]])
```

Looking at the generated c code, it appears that the call to ´casadi_triusolve´ uses the same pointer ´(&w1)´ for both the second argument (A matrix) and third argument (b and x vector).
```c
/* fun:(i0,i1[1x2])->(o0[1x2]) */
static int casadi_f0(const casadi_real** arg, casadi_real** res, casadi_int* iw, casadi_real* w, int mem) {
  casadi_real *w0=w+0, w1;
  /* #0: @0 = input[1][0] */
  casadi_copy(arg[1], 2, w0);
  /* #1: @1 = input[0][0] */
  w1 = arg[0] ? arg[0][0] : 0;
  /* #2: @0 = (@1\@0) */
  casadi_triusolve(casadi_s0, (&w1), (&w1), 0, 0, 2);
  /* #3: output[0][0] = @0 */
  casadi_copy(w0, 2, res[0]);
  return 0;
}
```
To correctly perform the operation, it seems like the correct call to triusolve should be `casadi_triusolve(casadi_s0, (&w1), w0, 0, 0, 2);`

This was found on casadi version 3.6.5

**Comment by jaeandersson** [comment]
Good find. It's trivial to fix: https://github.com/casadi/casadi/issues/3746

**Comment by andreasbell** [comment]
Fixed with casadi version 3.6.6
